code/obj/item/device/pda2/diagnostics.dm

```diff
@@ -481,3 +552,100 @@
 		f += 1
 	return f
 
+/proc/assoc_list_to_list(var/list/l)
+	var/list/keys = list()
+	var/list/vals = list()
+	for(var/key in l)
+		keys += key
+		vals += l[key]
+	return list(keys,vals)
+
+/proc/list_to_assoc_list(var/list/first,var/list/second)
+	var/list/result = list()
+	for(var/i = 1,i <= length(first),i++)
+		result[first[i]] = second[i]
+	return result
```

Comment:
@pali6 don't we already have these?

Reply:
i forgor 💀

Reply:
a quick code dive hasnt produced any results for these procs

Reply:
I don't think so. These should be in list.dm and not here certainly.

Reply:
Also the second proc should use . instead of a new var

Reply:
> Also the second proc should use . instead of a new var

i dont know what that does, could you explain? (also where to put?)

Reply:
It's the "default return value" variable. If you don't explicitly call return with some other value the value that actually gets returned is `.` . So here just replace result with . and remove the return statement. For more details consult the reference or DM By Example.

Reply:
done

Reply:
i think i should do resolve conversation now i think